src/pyodide/internal/snapshot.ts

```diff
@@ -262,8 +262,13 @@ function memorySnapshotDoImports(Module: Module): string[] {
   const deduplicatedModules = [...new Set(importedModules)];
 
   // Import the modules list so they are included in the snapshot.
-  if (deduplicatedModules.length > 0) {
-    simpleRunPython(Module, 'import ' + deduplicatedModules.join(','));
+  for (const mod of deduplicatedModules) {
+    try {
+      simpleRunPython(Module, `import ${mod}`);
+    } catch (e) {
+      continue;
+    }
```

Comment:
This feels fragile. Could we `continue` on only a specific kind of exception?

Also can you write a comment to explain exactly what kind of failure mode this solves?

Reply:
Fixed.